tests/test-tty

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
+EOF = b'\x04'
+
+
+def ttyflags(fd):
+    # see: http://www.gnu.org/software/libc/manual/html_mono/libc.html#Output-Modes
+    import termios as T
+    attrs = T.tcgetattr(fd)
+    attrs[1] &= ~T.OPOST  # don't munge output
+    attrs[3] &= ~T.ECHO  # don't echo input
+    T.tcsetattr(fd, T.TCSANOW, attrs)
+
+
+def child(dumb_init):
+    from os import execvp
+    execvp('timeout', ('timeout', '1', dumb_init, 'tac'))
+
+
+def parent(fd):
+    ttyflags(fd)
+    from os import read, write
+    assert write(fd, b'1\n2\n3\n') == 6
```

Comment:
oh, never mind, it's clear now.

I think docstrings on each of these methods would be really helpful though, I don't really know what `parent(fd)` means or is testing.